sympy/solvers/diophantine/diophantine.py

```diff
@@ -579,16 +601,17 @@ def solve(self, parameters=None, limit=None) -> DiophantineSolutionSet:
                 r = sqrt(discr)
                 u, v = symbols("u, v", integer=True)
                 eq = _mexpand(
-                    4*A*r*u*v + 4*A*D*(B*v + r*u + r*v - B*u) +
-                    2*A*4*A*E*(u - v) + 4*A*r*4*A*F)
+                    4*A*r*u*v + 4*A*D*(B*v + r*u + r*v - B*u) + 2*A*4*A*E*
+                    (u-v) + 4*A*r*4*A*F
```

Comment:
This is odd. I'd rather break on `+` than `*`.

Reply:
Well spotted, looks strange indeed... It seems that `split_before_arithmetic_operator = True` (c9054b11b6) helps here, and it's also more in line with black style:
```python
                eq = _mexpand(
                    4*A*r*u*v + 4*A*D*(B*v + r*u + r*v - B*u)
                    + 2*A*4*A*E*(u-v) + 4*A*r*4*A*F
                )
```